File: Tugas Besar/Ruangkelas.py
import requests
import json
class Ruangkelas:
    def __init__(self):
        self.__id=None
        self.__kode_ruang = None
        self.__nama_ruang = None
        self.__url = "http://a0832601.xsph.ru/AppTugasBesar/ruangkelas_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def kode_ruang(self):
        return self.__kode_ruang
        
    @kode_ruang.setter
    def kode_ruang(self, value):
        self.__kode_ruang = value
    @property
    def nama_ruang(self):
        return self.__nama_ruang
        
    @nama_ruang.setter
    def nama_ruang(self, value):
        self.__nama_ruang = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_kode_ruang(self, kode_ruang):
        url = self.__url+"?kode_ruang="+kode_ruang
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id_ruang']
            self.__kode_ruang = item['kode_ruang']
            self.__nama_ruang = item['nama_ruang']
        return data
    def simpan(self):
        payload = {
            "kode_ruang":self.__kode_ruang,
            "nama_ruang":self.__nama_ruang
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_kode_ruang(self, kode_ruang):
        url = self.__url+"?kode_ruang="+kode_ruang
        payload = {
            "kode_ruang":self.__kode_ruang,
            "nama_ruang":self.__nama_ruang
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_kode_ruang(self,kode_ruang):
        url = self.__url+"?kode_ruang="+kode_ruang
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text

File: Tugas Besar/Ruangkelas.php
<?php
//Simpanlah dengan nama file : Ruangkelas.php
require_once 'database.php';
class Ruangkelas 
{
    private $db;
    private $table = 'ruangkelas';
    public $kode_ruang = "";
    public $nama_ruang = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_kode_ruang(int $kode_ruang)
    {
        $query = "SELECT * FROM $this->table WHERE kode_ruang = $kode_ruang";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`kode_ruang`,`nama_ruang`) VALUES ('$this->kode_ruang','$this->nama_ruang')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET kode_ruang = '$this->kode_ruang', nama_ruang = '$this->nama_ruang' 
        WHERE id_ruang = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_kode_ruang($kode_ruang): int
    {
        $query = "UPDATE $this->table SET kode_ruang = '$this->kode_ruang', nama_ruang = '$this->nama_ruang' 
        WHERE kode_ruang = $kode_ruang";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id_ruang = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_kode_ruang($kode_ruang): int
    {
        $query = "DELETE FROM $this->table WHERE kode_ruang = $kode_ruang";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>